src/ansi-c/ansi_c_internal_additions.cpp

```diff
@@ -213,7 +213,17 @@ void ansi_c_internal_additions(std::string &code)
         code+="typedef long double __float128;\n";
     }
 
-    // On 64-bit systems, both gcc and clang have typedefs
+    if(
+      config.ansi_c.arch == "i386" || config.ansi_c.arch == "x86_64" ||
+      config.ansi_c.arch == "x32" || config.ansi_c.arch == "ia64")
```

Comment:
How has this subset of architectures been determined and/or why is it different from the `__float128` one?